subsys/net/ip/Kconfig

```diff
@@ -608,6 +608,14 @@ config NET_CONTEXT_RECV_PKTINFO
 	  This way user can get extra information about the received data in the
 	  socket.
 
+config NET_CONTEXT_RECV_HOPLIMIT
+	bool "Add receive PKTINFO support to net_context"
+	depends on NET_UDP
```

Comment:
The option should work for TCP sockets for IPv6, at least that is mentioned here https://www.ibm.com/docs/en/zos/2.4.0?topic=soadsiiil-options-that-provide-information-about-packets-that-have-been-received
although I did not spot mention of TCP in the RFC https://www.ietf.org/rfc/rfc3542.txt
But the `IP_RECVTTL` specifically mentions that stream socket is not supported.
So we could try to reflect this in the code by adding needed runtime checks there.

Reply:
I'd personally focus on datagrams only for now. That use case for stream sockets seems a bit odd.